### ISSUE TITLE
Domain-driven design

### ISSUE THREAD

**Issue by sehucke** [Garbanzo-recipes/garbanzo#6]
IMHO we should do some domain-driven design in a lean way first.

- [x] #7: Define the project vision and goals.
- [x] #7: Write down the central use cases (probably what we discussed last week ;-) )
- [ ] #8: Define the system architecture (see #1 and #3 ... but that are only two parts of a bigger story)
- [ ] #9: Create a domain model

**Comment by dlemper** [comment]
The first two parts are done from my point of viewas you described them in #7. if you agree, please check the two checkboxes above.
The architecture definitely needs more attention, thats true.
I‘m not sure why you think a domain model is needed here, can you elaborate on that?


<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by sehucke** [comment]
TL;DR: Because it is fun and I like models. :-)

Long story: Because I have a lot of thoughts in mind that I need to get in shape. I need to check how they align and how we build the data and features around them. It is more about the sharing and collaboration parts of the recipes/cookbooks, not about the recipes itself... I think the latter part is an easy one. :-)

Am 22. August 2018 00:06:30 MESZ schrieb Daniel Lemper <notifications@github.com>:

**Comment by dlemper** [comment]
Ok, feel free to create the domain models!

However I‘d like to start developing/prototyping as soon as possible and it won‘t be a problem to me to rewrite everything where a domain model differs from code. Is that ok for you, too?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by sehucke** [comment]
+1 for that! Prepare for rewriting everything! ;-P

OK, seriously: Prototyping is an excellent start to get a grip on the topic. :+1:  IMHO, we should start with a UI-only prototype and finish iteration 1 of the general system architecture in the meantime (especially the topic about supported backends).

**Comment by dlemper** [comment]
closed as it's stale for over a year
